api/src/main/java/ca/bc/gov/educ/api/grad/report/service/CommonService.java

```diff
@@ -845,6 +845,32 @@ private synchronized List<ReportGradStudentData> getReportGradStudentData(String
                 .block();
     }
 
+    public Integer countBySchoolOfRecordsAndReportType(List<String> schoolOfRecords, String reportType) {
+        Integer reportsCount = 0;
+        if(schoolOfRecords != null && !schoolOfRecords.isEmpty()) {
+            reportsCount += schoolReportsRepository.countBySchoolOfRecordsAndReportType(schoolOfRecords, reportType);
+        }
+        return reportsCount;
+    }
+
+    @Transactional
+    public Integer archiveSchoolReports(long batchId, List<String> schoolOfRecords, String reportType) {
+        Integer updatedReportsCount = 0;
+        Integer deletedReportsCount = 0;
+        Integer originalReportsCount = 0;
+        if(schoolOfRecords != null && !schoolOfRecords.isEmpty()) {
+            reportType = StringUtils.upperCase(StringUtils.endsWithIgnoreCase(reportType, "ARC") ? StringUtils.removeEndIgnoreCase(reportType, "ARC") : reportType);
+            String archivedReportType = StringUtils.upperCase(StringUtils.endsWith(reportType, "ARC") ? reportType : reportType + "ARC");
+            originalReportsCount += schoolReportsRepository.countBySchoolOfRecordsAndReportType(schoolOfRecords, reportType);
+            updatedReportsCount += schoolReportsRepository.archiveSchoolReports(schoolOfRecords, reportType, archivedReportType, batchId);
+            if(originalReportsCount.equals(updatedReportsCount)) {
```

Comment:
The logic here is right?   It seems School reports are updated as Archived, and then deleted.